sunpy/image/transform.py

```diff
@@ -130,20 +128,7 @@ def affine_transform(image, rmatrix, order=3, scale=1.0, image_center=None,
                           RuntimeWarning)
             adjusted_image = np.nan_to_num(adjusted_image)
```

Comment:
NaNs are casted to zeros in scipy and for higher order in scikit image, this function doesn't do it.